data-prepper-plugins/kafka-connect-plugins/README.md

```diff
@@ -0,0 +1,278 @@
+
+#### Collection
+
+| Option           | Required | Default | Type   | Description                                                                                                         |
+|------------------|----------|---------|--------|---------------------------------------------------------------------------------------------------------------------|
+| topic_prefix     | YES      |         | String | Unique name that identifies the connector.                                                                        |
+| collection_name  | YES      |         | String | The table name to ingest, using *database.collectionName* format.                                                     |
+
+
+### Credentials
+PlainText username and password
+```yaml
+credentials:
+    plaintext:
+      username: username
+      password: password
+```
+Leverage AWS Secret Manager for username and password
+```yaml
+credentials:
+  secret_manager:
+    region: us-east-1 #required aws region
+    secretId: secretId #required secret manager secretId
+    sts_role_arn: sts-role #optional
+```
+Note:
+* `sts_role_arn` (Optional) : The AWS STS role to assume for requests to Secret Manager. Defaults to null, which will use the [standard SDK behavior for credentials](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/credentials.html).
+
+### Kafka Cluster Configuration
+
+The Kafka Cluster Configuration must be set in data-prepper's configuration yaml.
+
+```yaml
+kafka_cluster_config:
+  bootstrap_servers:
+    - localhost:9092
+  encryption:
+    type: ssl
+    insecure: true
+  aws:
+    sts_role_arn: sts-role-arn
+    region: us-east-1
+    msk:
+      broker_connection_type: public
+      arn: msk-arn
+  authentication:
+    sasl:
+      aws_msk_iam: default
+```
+* `bootstrap_servers` (Optional) Required if `aws` and `msk` is not configured.
+* `encryption` (Required) https://opensearch.org/docs/latest/data-prepper/pipelines/configuration/sources/kafka/#encryption
+* `aws` (Optional) https://opensearch.org/docs/latest/data-prepper/pipelines/configuration/sources/kafka/#aws
+* `authentication` (Optional) https://opensearch.org/docs/latest/data-prepper/pipelines/configuration/sources/kafka/#authentication
+
+# Metrics
+
+## Kafka Connect Metrics
+Kafka Connect Worker Metrics
+
+- `task-count`: Number of tasks that have run in this worker
+- `connector-count`: The number of connectors that have run in this worker
+- `connector-startup-attempts-total`: Total number of connector startups that this worker has attempted
+- `connector-startup-success-total`: Total number of connector starts that succeeded
+- `connector-startup-failure-total`: Total number of connector starts that failed
+- `task-startup-attempts-total`: Total number of task startups that the worker has attempted
+- `task-startup-success-total`: Total number of task starts that succeeded
+- `task-startup-failure-total`: Total number of task starts that failed
+
+## Connector Metrics
```

Comment:
how is the metrics differentiated for each connector ? is there a prefix/suffix or other dimension ?

Reply:
It is using tags.
Eg.
pipeline_mongodb_source_record_active_count_avg{connector="prefix1.test.customers",serviceName="dataprepper",task="0",} 1341.987951807229